src/acl/ConnMark.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 1996-2018 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_ACLCONNMARK_H
+#define SQUID_ACLCONNMARK_H
+
+#include "acl/Acl.h"
+#include "ip/forward.h"
+#include "parser/Tokenizer.h"
+
+#include <map>
+
+namespace Acl {
+
+class ConnMark : public ACL
+{
+    MEMPROXY_CLASS(ConnMark);
+
+public:
+    /* ACL API */
+    virtual char const *typeString() const override;
+    virtual void parse() override;
+    virtual int match(ACLChecklist *checklist) override;
+    virtual SBufList dump() const override;
+    virtual bool empty() const override;
+
+private:
+    nfmark_t getNumber(Parser::Tokenizer &tokenizer, const SBuf &token);
+    std::map<nfmark_t, nfmark_t> marks;
```

Comment:
The type looks buggy! The new ACL does not _map_ marks to masks. An admin may, for example, configure the ACL with two mark/masks pairs that have the same mark but a different mask. This is especially likely in generated configurations where those pairs come from some external source. 

AFAICT, the new ACL uses a collection of mark/mask pairs. I suggest using std::vector for the iteration speed sake and to preserve admin-specified matching order. We could optimize further, including by merging some pairs, but I do _not_ think we should do that in this PR.